sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -292,31 +297,25 @@ class Dataset[T] private[sql](
       }
 
       // Create SeparateLine
-      val sep: String = colWidths.map("-" * _).addString(sb, "+", "+", "+\n").toString()
+      val sep: String = if (html) {
+        // Initial append table label
+        sb.append("<table border='1'>\n")
+        "\n"
+      } else {
+        colWidths.map("-" * _).addString(sb, "+", "+", "+\n").toString()
+      }
 
       // column names
-      rows.head.zipWithIndex.map { case (cell, i) =>
-        if (truncate > 0) {
-          StringUtils.leftPad(cell, colWidths(i))
-        } else {
-          StringUtils.rightPad(cell, colWidths(i))
-        }
-      }.addString(sb, "|", "|", "|\n")
-
+      appendRowString(rows.head, truncate, colWidths, html, true, sb)
       sb.append(sep)
 
       // data
-      rows.tail.foreach {
-        _.zipWithIndex.map { case (cell, i) =>
-          if (truncate > 0) {
-            StringUtils.leftPad(cell.toString, colWidths(i))
-          } else {
-            StringUtils.rightPad(cell.toString, colWidths(i))
-          }
-        }.addString(sb, "|", "|", "|\n")
+      rows.tail.foreach { row =>
+        appendRowString(row.map(_.toString), truncate, colWidths, html, false, sb)
```

Comment:
I think we need this toString, because the appendRowString method including both column names and data in original logic, which call `toString` in data part.

Reply:
I know this is not your change, but the `rows` is already `Seq[Seq[String]]` and the `row` is `Seq[String]`, so I think we can remove it.

Reply:
I see, the `cell.toString` has been called here. https://github.com/apache/spark/pull/21370/files/f2bb8f334631734869ddf5d8ef1eca1fa29d334a#diff-7a46f10c3cedbf013cf255564d9483cdR271
Got it, I'll fix this in next commit.